ee/app/engagement-dashboard/client/components/UsersTab/ActiveUsersSection.js

```diff
@@ -46,21 +46,23 @@ const ActiveUsersSection = () => {
 		});
 
 		const createPoints = () => Array.from({ length: moment(period.end).diff(period.start, 'days') + 1 }, (_, i) => createPoint(i));
-
-		const distributeValueOverPoints = (value, i, T, array, prev) => {
-			for (let j = 0; j < T; ++j) {
-				const k = i + j;
-
-				if (k >= array.length) {
-					continue;
-				}
-
+		const usersLists = data.month.reduce((map, dayData) => {
+			map[dayData.day] = dayData.usersList;
+			return map;
+		}, {});
+
+		const distributeValueOverPoints = (usersLists, currentIndex, usersList, i, T, array, prev) => {
+			const usersSet = new Set(usersList);
+			let j;
+			let k;
+			for (j = 0, k = i; j < T && k < array.length; ++k, ++j) {
 				if (k >= 0) {
-					array[k].y += value;
+					usersLists[currentIndex + j]?.forEach((userId) => usersSet.add(userId));
+					array[k].y = Math.max(usersSet.size, array[k].y);
```

Comment:
Hey @pierre-lehnen-rc and @tassoevan , I'm still not so sure whether WAU and MAU are being count correctly here 😵 . Any ideas?